### ISSUE TITLE
Spurious warning

### ISSUE THREAD

**Issue by cpitclaudel** [PyCQA/pylint#1099]
The following code produces a false positive:

``` python
def valid():
    for x in range(5):
        print((lambda _, x=x: x)(1))
valid()
```

```
$ pylint -r n /tmp/test.py
No config file found, using default configuration
************* Module test
C:  1, 0: Missing module docstring (missing-docstring)
C:  1, 0: Missing function docstring (missing-docstring)
W:  2, 8: Unused variable 'blah' (unused-variable)
```

This last warning is incorrect.

```
$ pylint --version
No config file found, using default configuration
pylint 1.6.1, 
astroid 1.4.7
Python 2.7.12 (default, Jul  1 2016, 15:12:24) 
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
I can reproduce it, thanks.
